core/src/main/java/org/kiwix/kiwixmobile/core/utils/SharedPreferenceUtil.kt

```diff
@@ -0,0 +1,177 @@
+import org.kiwix.kiwixmobile.core.CoreApp.Companion.instance
+import org.kiwix.kiwixmobile.core.NightModeConfig
+import org.kiwix.kiwixmobile.core.NightModeConfig.Mode.Companion.from
+import java.io.File
+import java.util.Locale
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Manager for the Default Shared Preferences of the application.
+ */
+@Singleton
+class SharedPreferenceUtil @Inject constructor(context: Context?) {
+  private val sharedPreferences: SharedPreferences =
+    PreferenceManager.getDefaultSharedPreferences(context)
+  val prefStorages = PublishProcessor.create<String>()
+  private val textZooms = PublishProcessor.create<Int>()
+  private val nightModes = PublishProcessor.create<NightModeConfig.Mode>()
+
+  val prefWifiOnly: Boolean
+    get() = sharedPreferences.getBoolean(PREF_WIFI_ONLY, true)
+
+  val prefHideToolbar: Boolean
+    get() = sharedPreferences.getBoolean(PREF_HIDE_TOOLBAR, true)
+
+  val prefIsFirstRun: Boolean
+    get() = sharedPreferences.getBoolean(PREF_IS_FIRST_RUN, true)
+
+  val prefFullScreen: Boolean
+    get() = sharedPreferences.getBoolean(PREF_FULLSCREEN, false)
+
+  val prefBackToTop: Boolean
+    get() = sharedPreferences.getBoolean(PREF_BACK_TO_TOP, false)
+
+  val prefNewTabBackground: Boolean
+    get() = sharedPreferences.getBoolean(PREF_NEW_TAB_BACKGROUND, false)
+
+  val prefExternalLinkPopup: Boolean
+    get() = sharedPreferences.getBoolean(PREF_EXTERNAL_LINK_POPUP, true)
+
+  val prefLanguage: String
+    get() = sharedPreferences.getString(PREF_LANG, "")?.toString() ?: Locale.ROOT.toString()
+
+  val prefStorage: String
+    get() {
+      val storage = sharedPreferences.getString(PREF_STORAGE, null)
+      if (storage == null) {
+        val defaultStorage = defaultStorage()
+        putPrefStorage(defaultStorage)
+        return defaultStorage().also(::putPrefStorage)
+      } else if (!File(storage).exists()) {
+        return defaultStorage()
+      }
+      return storage
+    }
+
+  private fun defaultStorage(): String =
+    getExternalFilesDirs(instance, null)[0]?.path ?: instance.filesDir.path
+
+  fun getPrefStorageTitle(defaultTitle: String): String =
+    sharedPreferences.getString(PREF_STORAGE_TITLE, "")?.toString() ?: defaultTitle
+
+  fun putPrefLanguage(language: String) =
+    sharedPreferences.edit { putString(PREF_LANG, language) }
+
+  fun putPrefIsFirstRun(isFirstRun: Boolean) =
+    sharedPreferences.edit { putBoolean(PREF_IS_FIRST_RUN, isFirstRun) }
+
+  fun putPrefWifiOnly(wifiOnly: Boolean) =
+    sharedPreferences.edit { putBoolean(PREF_WIFI_ONLY, wifiOnly) }
+
+  fun putPrefStorageTitle(storageTitle: String) =
+    sharedPreferences.edit { putString(PREF_STORAGE_TITLE, storageTitle) }
+
+  fun putPrefStorage(storage: String) {
+    sharedPreferences.edit { putString(PREF_STORAGE, storage) }
+    prefStorages.onNext(storage)
+  }
+
+  fun getPrefStorages(): Flowable<String> = prefStorages.startWith(prefStorage)
```

Comment:
This method is still unused. 

It is not being used because of a visibility conflict, I don't know how much you want me to tell you and how much detective work you want to do on your own

Reply:
It isn't like I overlooked the fact of the concerned function not being used, i did limit the scope of `prefStorages` in the kotlin version which automatically was increased to global one, but with the change being made, the `Fat32Checker.kt` was unable to access this function which lead build to failing so i reverted the visibility for the time being and pushed here so that it may become easier for me to look the differences and change accordingly and this was the solo reason after pushing those commits you were neither tagged nor were requested for a review, so now onwards considering this seriously, i'll mark the PR as draft and continue working on it until its finalized or near around that. 
Regrets for the inconvenience caused.